File: win-prog/ProgWinVB/Lines Curves and Area Fills/DrawHouse/DrawHouse.vb
'------------------------------------------
' DrawHouse.vb (c) 2002 by Charles Petzold
'------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class DrawHouse
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New DrawHouse())
    End Sub

    Sub New()
        Text = "Draw a House in One Line"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        grfx.DrawLines(New Pen(clr), _
                       New Point() _
                       { _
                           New Point(cx \ 4, 3 * cy \ 4), _
                           New Point(cx \ 4, cy \ 2), _
                           New Point(cx \ 2, cy \ 4), _
                           New Point(3 * cx \ 4, cy \ 2), _
                           New Point(3 * cx \ 4, 3 * cy \ 4), _
                           New Point(cx \ 4, cy \ 2), _
                           New Point(3 * cx \ 4, cy \ 2), _
                           New Point(cx \ 4, 3 * cy \ 4), _
                           New Point(3 * cx \ 4, 3 * cy \ 4) _
                       })
    End Sub
End Class

File: win-prog/ProgWinCS/Lines Curves and Area Fills/DrawHouse/DrawHouse.cs
//----------------------------------------
// DrawHouse.cs © 2001 by Charles Petzold
//----------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class DrawHouse: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new DrawHouse());
     }
     public DrawHouse()
     {
          Text = "Draw a House in One Line";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          grfx.DrawLines(new Pen(clr),
                         new Point[] 
                         {
                         new Point(    cx / 4, 3 * cy / 4), // Lower left
                         new Point(    cx / 4,     cy / 2),
                         new Point(    cx / 2,     cy / 4), // Peak
                         new Point(3 * cx / 4,     cy / 2),
                         new Point(3 * cx / 4, 3 * cy / 4), // Lower right
                         new Point(    cx / 4,     cy / 2),
                         new Point(3 * cx / 4,     cy / 2),
                         new Point(    cx / 4, 3 * cy / 4), // Lower left
                         new Point(3 * cx / 4, 3 * cy / 4)  // Lower right
                         });
     }
}
